OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -166,10 +166,21 @@ internal static class DreamProcNativeRoot {
 
         [DreamProc("arctan")]
         [DreamProcParameter("A", Type = DreamValueTypeFlag.Float)]
+        [DreamProcParameter("x", Type = DreamValueTypeFlag.Float)]
+        [DreamProcParameter("y", Type = DreamValueTypeFlag.Float)]
         public static DreamValue NativeProc_arctan(NativeProc.Bundle bundle, DreamObject? src, DreamObject? usr) {
-            bundle.GetArgument(0, "A").TryGetValueAsFloat(out float a);
-            double atan = Math.Atan(a);
-
+            DreamValue A = bundle.GetArgument(0, "A");
+            DreamValue x = bundle.GetArgument(0, "x");
+            DreamValue y = bundle.GetArgument(1, "y");
+            double atan;
+            if(y == DreamValue.Null){
+                A.TryGetValueAsFloat(out float Afloat);
+                atan = Math.Atan(Afloat);
+            } else {
+                x.TryGetValueAsFloat(out float xfloat);
+                y.TryGetValueAsFloat(out float yfloat);
+                atan = Math.Atan2(yfloat, xfloat);
+            }
```

Comment:
`arctan(30)` and `arctan(30, null)` give 2 different results. Instead check if `bundle.Arguments.Length` is 1 or 2.
```suggestion
            if (bundle.Arguments.Length == 1) {
                atan = Math.Atan(x);
            } else {
                bundle.GetArgument(1, "y").TryGetValueAsFloat(out float y);
                atan = Math.Atan2(y, x);
            }
```